cmd/config/internal/commands/cmdset.go

```diff
@@ -75,16 +79,25 @@ func initSetterVersion(c *cobra.Command, args []string) error {
 }
 
 func (r *SetRunner) preRunE(c *cobra.Command, args []string) error {
+	valueFlagSet := isFlagSet("values", c)
+
+	if valueFlagSet && len(args) > 2 {
+		return errors.Errorf("value should set either from flag or arg")
+	}
+
 	if len(args) > 1 {
 		r.Perform.Name = args[1]
 		r.Lookup.Name = args[1]
 	}
-	if len(args) > 2 {
+
+	if isFlagSet("values", c) && len(r.Values) == 1 {
```

Comment:
Do we need this check?  If we do should it be > 1?

Reply:
removed.